firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/BreathingCircuit/AlarmMuteService.cpp

```diff
@@ -0,0 +1,27 @@
+/*
+ * BreathingCircuit.tpp
+ *
+ *  Created on: June 6, 2020
+ *      Author: Ethan Li
+ */
+
+#include "Pufferfish/Driver/BreathingCircuit/AlarmMuteService.h"
+
+// AlarmMuteService
+
+namespace Pufferfish::Driver::BreathingCircuit {
+
+void AlarmMuteService::transform(
+    AlarmMute &alarm_mute, const AlarmMuteRequest &alarm_mute_request) {
+  alarm_mute.active = alarm_mute_request.active;
```

Comment:
Right now the simulator backend returns early if there are no active alarms, while the firmware never returns early. Whichever approach we take, I think the Alarm Mute Service should have consistent behavior between the backend and firmware.

Reply:
Yeah I removed the check as it was the case like you had suggested that active_alarms_ in firmware was always empty and it returned early, I'll remove it in the simulator too now !